File: Atividades/att3/Python/main.py
import merge_sort as m, bubble_sort as b
import time, tracemalloc

unsorted = list()
element = "text"
value = 0

with open("../arq.txt", "r") as lista:
    while element != "":
        element = lista.readline()
        try:
            value = int(element)
        except:
            break
        unsorted.append(value)

tracemalloc.start()
inic = time.time()
sorted_bubble = b.bubble_sort(list(unsorted))
fim = time.time()
tempo = round( (fim - inic), 2)
mem = tracemalloc.get_traced_memory()[1]
tracemalloc.stop()
print(f"tempo do bubble sort :{tempo}\n memoria utilizada {mem}")

tracemalloc.start()
inic = time.time()
sorted_merge = m.merge_sort(list(unsorted))
fim = time.time()
tempo = round( (fim - inic) * 1000, 2)
mem = tracemalloc.get_traced_memory()[1]
tracemalloc.stop()

print(f"tempo do merge sort {tempo}\n memoria utilizada {mem}")

File: Atividades/att3/Go/main.go
package main

import (
	sorters "att3/mod"
	"fmt"
	"io"
	"os"
	"runtime"
	"time"
)

// literalmente um copiar-colar do grok
// getMemStats captura estatísticas de memória
func getMemStats() runtime.MemStats {
	var memStats runtime.MemStats
	runtime.ReadMemStats(&memStats)
	return memStats
}

func ReadFile(filePath string) (lista []int) {
	fd, err := os.Open(filePath)
	if err != nil {
		panic(err)
	}
	var line int
	for {
		_, err := fmt.Fscanf(fd, "%d\n", &line)

		if err != nil {
			fmt.Print(err)
			if err == io.EOF {
				return
			}
			panic(fmt.Sprint("scan failed %s, %v", filePath, err))
		}
		lista = append(lista, line)
	}

}

func main() {

	lista := ReadFile("../arq.txt")

	inicioBubble := getMemStats()
	start := time.Now()
	sorters.Bubble_sort(lista)
	tempoBubble := time.Since(start)
	memBubble := getMemStats().TotalAlloc - inicioBubble.TotalAlloc

	inicioMerge := getMemStats()
	start = time.Now()
	sorters.MergeSort(lista)
	tempoMerge := time.Since(start)
	memMerge := getMemStats().TotalAlloc - inicioMerge.TotalAlloc

	fmt.Printf("tempo decorrido bubble\n: %v\n", tempoBubble)
	fmt.Printf("tempo decorrido Merge\n: %v\n\n", tempoMerge)

	fmt.Printf("memória alocada bubble\n: %v\n", memBubble)
	fmt.Printf("memória alocada Merge\n: %v\n\n", memMerge)

}
